src/Interpreters/InterpreterCreateFormatSchemaQuery.cpp

```diff
@@ -0,0 +1,57 @@
+#include <Access/Common/AccessRightsElement.h>
+#include <Access/Common/AccessType.h>
+#include <Formats/FormatSchemaFactory.h>
+#include <Interpreters/Context.h>
+#include <Interpreters/InterpreterCreateFormatSchemaQuery.h>
+#include <Parsers/ASTCreateFormatSchemaQuery.h>
+
+namespace DB
+{
+
+namespace
+{
+std::string_view trim_whitespaces_at_both_ends(const String & str)
+{
+    std::string_view ret;
+    auto begin = std::find_if_not(str.begin(), str.end(), [](auto ch) { return std::isspace(ch); });
+    if (begin == str.end())
+        return ret;
+    auto end = std::find_if_not(str.rbegin(), str.rend(), [](auto ch) { return std::isspace(ch); });
+    size_t size = end.base() - begin;
+    return std::string_view{begin.base(), size};
+}
+
+}
+
+BlockIO InterpreterCreateFormatSchemaQuery::execute()
+{
+    auto & create_format_schema_query = query_ptr->as<ASTCreateFormatSchemaQuery &>();
+
+    if (!create_format_schema_query.cluster.empty())
+        throw Exception(ErrorCodes::NOT_IMPLEMENTED, "ON CLUSTER is not implemented for CREATE FORMAT SCHEMA");
+
+    auto body = create_format_schema_query.getSchemaBody();
+    auto trimmed_body = trim_whitespaces_at_both_ends(body);
+    if (trimmed_body.empty())
```

Comment:
we can rename `trim_whitespaces_at_both_ends` to `trimToView`

Reply:
I renamed it to just `trim`, because I think `ToView` is redundant, the function signature already says it returns a view.